jwst/group_scale/group_scale.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import division
+
+#
+#  Module for rescaling grouped data for NFRAME
+#  not equal to a power of 2
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def do_correction(input_model):
+    """
+    Short Summary
+    -------------
+    Rescales all groups in an exposure by FRMDIVSR/NFRAMES, to
+    account for incorrect scaling when on-board frame averaging
+    is done using a value of NFRAMES that is not a power of 2.
+    The divisor for the on-board averaging is always a power of
+    2, so if the number of frames coadded is not a power of 2,
+    the result is incorrect and needs to be rescaled.
+
+    Parameters
+    ----------
+    input_model: data model object
+        science data to be corrected
+
+    Returns
+    -------
+    output_model: data model object
+        rescaled science data
+
+    """
+
+    # Get the meta data values that we need
+    nframes = input_model.meta.exposure.nframes
+    frame_divisor = input_model.meta.exposure.frame_divisor
+    if (nframes is None) or (frame_divisor is None):
```

Comment:
The "nframes is a power of 2" check is done up in the group_scale_step function, but was not replicated here. I sort of figured that this lower-level function would only be called directly by folks who know what they're doing and hence the meta data would be complete and accurate at that point, and if it isn't just punt without attempting any heroic measures. But you never know ...